tst/com/amazon/corretto/crypto/provider/test/AesKeyWrapTest.java

```diff
@@ -46,50 +47,111 @@
 
     @ParameterizedTest
     @MethodSource("sizes")
-    public void roundtrip(int aeskeySize, int secretSize) throws Exception {
+    public void roundtripNativeSameCipher(int aesKeySize, int secretSize) throws Exception {
+        roundtrip(aesKeySize, secretSize, TestUtil.NATIVE_PROVIDER, null, true);
+    }
+
+    @ParameterizedTest
+    @MethodSource("sizes")
+    public void roundtripNativeNewCipher(int aesKeySize, int secretSize) throws Exception {
+        roundtrip(aesKeySize, secretSize, TestUtil.NATIVE_PROVIDER, TestUtil.NATIVE_PROVIDER, false);
+    }
+
+    @ParameterizedTest
+    @MethodSource("sizes")
+    public void roundtripNative2BC(int aesKeySize, int secretSize) throws Exception {
+        roundtrip(aesKeySize, secretSize, TestUtil.NATIVE_PROVIDER, TestUtil.BC_PROVIDER, false);
+    }
+
+    @ParameterizedTest
+    @MethodSource("sizes")
+    public void roundtripBC2native(int aesKeySize, int secretSize) throws Exception {
+        roundtrip(aesKeySize, secretSize, TestUtil.BC_PROVIDER, TestUtil.NATIVE_PROVIDER, false);
+    }
+
+    // TODO [childw] add new compatibility test cases for JCE providers that skip if < Java17
+    // TODO [childw] add parameter for SecretKey type, parameterize that out across EC, RSA, etc.
+    //               bonus points if we can auto-detect valid key sizes and filter the overarching
+    //               secret key parameter sizes into these parameterized tests for particular key sizes.
+    private void roundtrip(int aesKeySize, int secretSize, Provider wrappingProvider, Provider unwrappingProvider, boolean reuseCipher) throws Exception {
         final SecureRandom sr = new SecureRandom();
-        byte[] keyBytes = new byte[aeskeySize];
+        byte[] keyBytes = new byte[aesKeySize];
         sr.nextBytes(keyBytes);
         final SecretKey key = new SecretKeySpec(keyBytes, "AES");
 
         byte[] secretBytes = new byte[secretSize];
         sr.nextBytes(secretBytes);
         final SecretKey secret = new SecretKeySpec(secretBytes, "Generic");
 
-        Cipher c = Cipher.getInstance("AES/KWP/NoPadding", TestUtil.NATIVE_PROVIDER);
+        Cipher c = getCipher(wrappingProvider);
         c.init(Cipher.WRAP_MODE, key, sr);
         byte[] wrapped = c.wrap(secret);
         assertFalse(Arrays.equals(secretBytes, wrapped));
+        if (!reuseCipher) {
+            c = getCipher(unwrappingProvider);
+        } else {
+            assertTrue(unwrappingProvider == null);
+        }
         c.init(Cipher.UNWRAP_MODE, key, sr);
         Key unwrapped = c.unwrap(wrapped, "Generic", Cipher.SECRET_KEY);
-        //System.out.println("KEY:    " + Arrays.toString(key.getEncoded()));
-        //System.out.println("SECRET: " + Arrays.toString(secret.getEncoded()));
-        //System.out.println("WRAPPD: " + Arrays.toString(wrapped));
-        //System.out.println("UNWRAP: " + Arrays.toString(unwrapped.getEncoded()));
         assertTrue(Arrays.equals(secret.getEncoded(), unwrapped.getEncoded()));
```

Comment:
NitPick: [assertArraysHexEquals](https://github.com/corretto/amazon-corretto-crypto-provider/blob/3ad0ce7f46d4c0f5ef13ae7868502c10319af0de/tst/com/amazon/corretto/crypto/provider/test/TestUtil.java#L67) is your friend.